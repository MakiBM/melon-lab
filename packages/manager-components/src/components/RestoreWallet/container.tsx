import { withFormik } from 'formik';
import { compose, withHandlers } from 'recompose';
import * as Yup from 'yup';
import RestoreWallet from './index';

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.initialValues }),
  validationSchema: Yup.object().shape({
    mnemonic: Yup.string().required('mnemonic is required.'),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) => {
    if (form.props.onSubmit) {
      form.props.onSubmit(values);
    }
  },
});

const withFormHandler = compose(
  withHandlers({
    onChange: props => (values, event) => {
      props.setFieldValue(event.target.name, values.value);
    },
  }),
);

export default compose(
  withFormValidation,
  withFormHandler,
)(RestoreWallet);
