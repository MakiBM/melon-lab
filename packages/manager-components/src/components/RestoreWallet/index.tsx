import React, { StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';

import styles from './styles.css';

interface FormValues {
  mnemonic: string;
}

export interface RestoreWalletProps {
  handleSubmit?: () => void;
  handleBlur?: () => void;
  onChange?: () => void;
  values?: FormValues;
  error?: string;
  touched?: any;
  errors?: any;
}

export const RestoreWallet: StatelessComponent<RestoreWalletProps> = ({
  handleSubmit,
  handleBlur,
  onChange,
  error,
  values,
  touched,
  errors,
}) => {
  return (
    <div className="restore-wallet">
      <style jsx>{styles}</style>
      <h3>Confirm mnemonic</h3>
      <p>Please type your 12-words mnemonic:</p>
      {touched.mnemonic &&
        error && <p className="restore-wallet__error">{error}</p>}
      <Form>
        <div className="restore-wallet__input">
          <Input
            onChange={onChange}
            onBlur={handleBlur}
            value={values.mnemonic}
            required={true}
            name="mnemonic"
            type="text"
            placeholder="mnemonic"
            error={touched.mnemonic && errors.mnemonic}
          />
        </div>
        <Button type="submit" onClick={handleSubmit} style="secondary">
          Import
        </Button>
      </Form>
    </div>
  );
};

export default RestoreWallet;
