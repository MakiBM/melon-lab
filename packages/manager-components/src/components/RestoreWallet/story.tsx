import { storiesOf } from '@storybook/react';
import React from 'react';
import RestoreWallet from './container';

const data = {
  initialValues: {
    mnemonic: '',
  },
  error: '',
};

storiesOf('Components|Restore Wallet', module).add('Default', () => {
  return <RestoreWallet {...data} />;
});
