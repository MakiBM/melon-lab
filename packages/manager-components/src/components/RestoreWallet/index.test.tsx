import React from 'react';
import RestoreWallet from './index';

const data = {
  initialValues: {
    mnemonic: '',
  },
  error: '',
  touched: {
    mnemonic: false,
  },
  values: {
    mnemonic: '',
  }
};

describe('RestoreWallet', () => {
  const defaultElement = <RestoreWallet {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
