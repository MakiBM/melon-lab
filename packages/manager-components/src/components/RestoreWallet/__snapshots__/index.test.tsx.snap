// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestoreWallet should render correctly 1`] = `
<div
  className="jsx-__scopedHash restore-wallet"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Confirm mnemonic
  </h3>
  <p
    className="jsx-__scopedHash"
  >
    Please type your 12-words mnemonic:
  </p>
  <Form>
    <div
      className="jsx-__scopedHash restore-wallet__input"
    >
      <Input
        error={false}
        name="mnemonic"
        placeholder="mnemonic"
        required={true}
        type="text"
        value=""
      />
    </div>
    <Button
      style="secondary"
      type="submit"
    >
      Import
    </Button>
  </Form>
</div>
`;
